### PR TITLE
feat: Add functional-sourceDirectory

### PR DESCRIPTION
**Pull request by yoshwata**
## Context

Add a functional test for the [Source Directory](https://docs.screwdriver.cd/user-guide/configuration/sourceDirectory.html)

## Objective

This PR adds a functional test for the source directory.
The repository used for testing was created below:
https://github.com/screwdriver-cd-test/functional-sourceDirectory

## References

https://github.com/screwdriver-cd/screwdriver/issues/1393

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: features/trigger.feature [hunk 1]
```diff
@@ -132,3 +132,12 @@ Feature: Remote Trigger
...
 
+    @sourceDirectory
+    Scenario: sourceDirectory
+      Given an existing pipeline on branch "master" setting source directory "directory1" with the workflow jobs:
+            | job       | requires  |
+            | job1      | ~commit   |
+            | job2      | ~commit   |
+      When a new file is added to the "directory1" directory of the "master" branch
+      Then a new build from "job1" should be created to test that change
+      And a new build from "job2" should not be created to test that change
```
Comment (yoshwata) [inline features/trigger.feature:143]: should remove

File: features/support/world.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ function ensurePipelineExists(config) {
             .then(response => {
                 this.jwt = response.body.token;
 
-                return this.createPipeline(config.repoName, branch);
+                return this.createPipeline(config.repoName, branch, config.rootDir);
```
Comment (tkyi) [inline features/support/world.js:59]: can you add `config.rootDir` to jsdoc too

Comment (yoshwata) [reply features/support/world.js:60]: @tkyi Thank you. I added.

**Review by ibu1224 — APPROVED** [review]

**Review by y-oksaku — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Comment by tkyi** [comment]
I renamed default branch on https://github.com/screwdriver-cd-test/functional-sourceDirectory from `main` to `master`

**Pull request merged**